core/src/com/unciv/logic/automation/civilization/DeclareWarTargetAutomation.kt

```diff
@@ -0,0 +1,125 @@
+package com.unciv.logic.automation.civilization
+
+import com.unciv.logic.civilization.Civilization
+import com.unciv.logic.civilization.diplomacy.DiplomacyFlags
+import com.unciv.logic.civilization.diplomacy.RelationshipLevel
+import com.unciv.logic.trade.TradeLogic
+import com.unciv.logic.trade.TradeOffer
+import com.unciv.logic.trade.TradeRequest
+import com.unciv.logic.trade.TradeType
+import com.unciv.ui.screens.victoryscreen.RankingType
+
+object DeclareWarTargetAutomation {
+
+    /**
+     * Chooses a target civilization along with a plan of attack.
+     * Note that this doesn't guarantee that we will declare war on them immediatly, or that we will end up declaring war at all.
+     */
+    fun chooseDeclareWarTarget(civInfo: Civilization, civAttackMotivations: List<Pair<Civilization, Int>>) {
+        val highestValueTargets = civAttackMotivations.sortedByDescending { it.first.getStatForRanking(RankingType.Score) }
+
+        for (target in highestValueTargets) {
+            if (tryDeclareWarWithPlan(civInfo, target.first, target.second)) 
+                return // We have successfully found a plan and started executing it!
+        }
+    }
+
+    /**
+     * Determines a war plan against this [target] and executes it if able.
+     */
+    private fun tryDeclareWarWithPlan(civInfo: Civilization, target: Civilization, motivation: Int): Boolean {
+
+        if (!target.isCityState()) {
+            if (motivation > 5 && tryTeamWar(civInfo, target, motivation)) return true
+
+            if (motivation >= 15 && tryJoinWar(civInfo, target, motivation)) return true
+        }
+
+        if (motivation >= 50 && declareWar(civInfo, target, motivation)) return true
+
+        if (motivation >= 15 && prepareWar(civInfo, target, motivation)) return true
+
+        return false
+    }
+
+    /**
+     * The safest option for war is to invite a new ally to join the war with us.
+     * Together we are stronger and are more likely to take down bigger threats.
+     */
+    private fun tryTeamWar(civInfo: Civilization, target: Civilization, motivation: Int): Boolean {
```

Comment:
The AI might be a little to prone to initiating team wars in this version. It is however a good deal since they it also builds relationships with other civilizations.